packages/terra-layout/src/_LayoutSlidePanel.jsx

```diff
@@ -163,7 +163,7 @@ class LayoutSlidePanel extends React.Component {
           {panelContent}
         </div>
         <OverlayContainer className={cx('content')}>
-          <Overlay isRelativeToContainer onRequestClose={onToggle} isOpen={isOverlayOpen} backgroundStyle={overlayBackground} />
+          <Overlay isRelativeToContainer onRequestClose={onToggle} isOpen={isOverlayOpen} backgroundStyle={overlayBackground} zIndex="6000" />
```

Comment:
Worked with @neilpfeiffer on this change. This resolves on overlay issue between the layout slide-panel overlay and the z-index introduced on the application header.